django/db/backends/mysql/schema.py

```diff
@@ -34,10 +34,16 @@ def quote_value(self, value):
 
     def skip_default(self, field):
         """
-        MySQL doesn't accept default values for longtext and longblob
-        and implicitly treats these columns as nullable.
+        MySQL doesn't accept default values for TEXT and BLOB types, and
+        implicitly treats these columns as nullable.
         """
-        return field.db_type(self.connection) in {'longtext', 'longblob'}
+        db_type = field.db_type(self.connection)
+        return (
+            db_type is not None and
+            db_type.lower() in {
+                'tinytext', 'tinyblob', 'text', 'blob', 'mediumtext', 'mediumblob', 'longtext', 'longblob'
```

Comment:
I think it would be most readable to format as 
```
db_type.lower() in {
    'blob', 'tinyblob',  'mediumblob', 'longblob',
    'text', 'tinytext',  'mediumtext', 'longtext',
}
```